metrics-influxdb/src/main/java/io/dropwizard/metrics5/influxdb/InfluxDbReporter.java

```diff
@@ -0,0 +1,517 @@
+        writeFieldIfEnabled(P999,   snapshot.get999thPercentile());
+        if (hasValues()) {
+            writeTimestampMillis(timestamp);
+            reportLine();
+        }
+    }
+    
+    
+    private void reportMetered(MetricName name, Metered meter, long timestamp) throws IOException {
+        writeMeasurement(name);
+        writeMeteredFieldsIfEnabled(meter);
+        if (hasValues()) {
+            writeTimestampMillis(timestamp);
+            reportLine();
+        }
+    }
+
+    private void writeMeteredFieldsIfEnabled(Metered meter) {
+        writeFieldIfEnabled(COUNT,     meter.getCount());
+        writeFieldIfEnabled(SUM,       meter.getSum());
+        writeFieldIfEnabled(M1_RATE,   convertRate(meter.getOneMinuteRate()));
+        writeFieldIfEnabled(M5_RATE,   convertRate(meter.getFiveMinuteRate()));
+        writeFieldIfEnabled(M15_RATE,  convertRate(meter.getFifteenMinuteRate()));
+        writeFieldIfEnabled(MEAN_RATE, convertRate(meter.getMeanRate()));
+    }
+    
+    private void reportCounter(MetricName name, Counter counter, long timestamp) throws IOException {
+        writeMeasurement(name);
+        writeFieldIfEnabled(COUNT, counter.getCount());
+        if (hasValues()) {
+            writeTimestampMillis(timestamp);
+            reportLine();
+        }
+    }
+
+    private void reportGauge(MetricName name, Gauge<?> gauge, long timestamp) throws IOException {
+        writeMeasurement(name);
+        Object value = gauge.getValue();
+        if (value != null) {
+            writeField(VALUE);
+            if (value instanceof Number) {
+                Number number = (Number) value;
+                if (number instanceof Long || 
+                        number instanceof Integer || 
+                        number instanceof Short || 
+                        number instanceof Byte) {
+                    writeFieldValue(number.longValue());
+                } else {
+                    writeFieldValue(number.doubleValue());
+                }
+            } else if (value instanceof Boolean) {
+                writeFieldValue(((Boolean)value));
+            } else {
+                writeFieldValue(value.toString());
+            }
+        }
+        writeTimestampMillis(timestamp);
+        reportLine();
+    }    
+    
+    private void reset() {
+        str.delete(0, str.length());
+    }
+    
+    private void writeMeasurement(MetricName name) {
+        reset();
+        char[] nameChars = encodedNameCache.get(name);
+        if (nameChars != null) {
+            str.append(nameChars);
+        } else {
+            writeMeasurementNoCache(name);
+            nameChars = new char[str.length()];
+            str.getChars(0, str.length(), nameChars, 0);
+            encodedNameCache.put(name, nameChars);
+        }
+        
+        str.append(' ');
+        firstField = true;
+    }
+    
+    private void writeMeasurementNoCache(MetricName name) {
```

Comment:
If we cache names of metrics, I think it's better to store them as strings instead of char arrays. Also I don't think it makes much sense to cache a value and at the time write it to the current `StringBuilder`. I would expect that we just read the data from the cache and if the key is not present, we load it: `str.append(encodedNameCache.computeIfAbsent(name, this::toInfluxMetricName))`